MAIN-143: Docs for Creating Tic-Tac-Toe / Backend (#6)
turnbasedgames/turnbasedgames.github.io: docs/backend.md

---
title: Understanding the Backend
---

## Overview

The "backend" for all games is compromised of four functions found in the highest level **index.js** file. Currently, for this file to work you cannot import any outside functions - all of your code must be in this file.

## Objects

### BoardGame

```json
{
  "joinable": true,
  "finished": false,
  "players": "[]",
  "version": 0,
  "state": {}
}
```

A JSON object provided to you that contains information about the current board game state.

#### joinable: *boolean*

Initially true, can be modified. 

> If true, new users will be able to join this game instance. If false, this game instance will not be included in the matchmaking queue and new players will be blocked from joining a private room with this game instance.

Reviewer: Is the goal to be more descriptive here for the `false` case? Also we do not necessarily use a queue for matchmaking: 
```suggestion
If true, new users will be able to join this game instance. If false, new users can not join this game instance via a private room or matchmaking.
```

Author: oh yes much better
